### PR TITLE
OCPBUGS-35205: Add hypershift-cluster-version-operator image to release providers

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:
- allows CVO image to use registry overrides

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: support/releaseinfo/static_provider.go [hunk 1]
```diff
@@ -29,6 +29,14 @@ func (p *StaticProviderDecorator) Lookup(ctx context.Context, image string, pull
 	if err != nil {
 		return nil, err
 	}
+
+	releaseImage.Spec.Tags = append(releaseImage.Spec.Tags, imageapi.TagReference{
```
Comment (csrwng) [inline support/releaseinfo/static_provider.go:33]: why just not add an image [here](https://github.com/openshift/hypershift/blob/main/control-plane-operator/main.go#L357-L364)?

Comment (muraee) [reply support/releaseinfo/static_provider.go:33]: at that point we don't know what the release image is yet.

Comment (csrwng) [reply support/releaseinfo/static_provider.go:33]: it is available to the process: https://github.com/openshift/hypershift/blob/e6407b07158d3abf317f02021751c55e44d2166b/control-plane-operator/main.go#L417

Comment (muraee) [reply support/releaseinfo/static_provider.go:33]: nice, do you know since when we set this env variable? I am asking if we want backport this.

Comment (csrwng) [reply support/releaseinfo/static_provider.go:33]: Looks like long enough ago that it should be in all supported versions 
https://github.com/openshift/hypershift/pull/1023

Comment (muraee) [reply support/releaseinfo/static_provider.go]: updated

File: control-plane-operator/controllers/hostedcontrolplane/cvo/params.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ func NewCVOParams(hcp *hyperv1.HostedControlPlane, releaseImageProvider *imagepr
 		CLIImage:                releaseImageProvider.GetImage("cli"),
 		AvailabilityProberImage: releaseImageProvider.GetImage(util.AvailabilityProberImageName),
 		ControlPlaneImage:       util.HCPControlPlaneReleaseImage(hcp),
-		ReleaseImage:            hcp.Spec.ReleaseImage,
+		ReleaseImage:            releaseImageProvider.GetImage("hypershift-cluster-version-operator"),
```
Comment (csrwng) [inline control-plane-operator/controllers/hostedcontrolplane/cvo/params.go:31]: Just a couple more things:
- can we just rename this to "cluster-version-operator" ?
- could we fall back to hcp.Spec.ReleaseImage if we cannot get the one from the releaseImageProvider? (I'm thinking of local development use case where we may not be setting the env var)

Comment (muraee) [reply control-plane-operator/controllers/hostedcontrolplane/cvo/params.go]: updated.

**Pull request merged**